### DISCUSSION TITLE
Myenergi Integration and Octopus Intelligent question

### DISCUSSION THREAD

**Discussion by Moondevil-ha** [CJNE/ha-myenergi#605] [Q&A]
Hi,
I don't want to raise an issue, as I'm sure I can get round this with some change to API polling, but recent changes in the Octopus Intelligent dispatch scheduling have lead me to get numerous "we're unable to control your charger" messages. I've tried a couple of things, but the only thing that cures this is removing (or disabling) the myenergi integration.

Do you think I may be hitting some API limit, so maybe change the polling to 180secs rather than 60? I re-enabled it last night and the errors started back again, so I've removed temporarily. Maybe it's coincidence.

Any other ideas?

**Comment by G6EJD** [comment]
Once you have handed control to Octopus you can no-longer use the HA MyEnergi integration to control your Zappi. This is a well known and widely discussed topic in the MyEnergi forum.

**Comment by LeiChat** [comment]
I use https://github.com/[megakid/ha_octopus_intelligent](https://github.com/megakid/ha_octopus_intelligent) rather than https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy. Only because I was not bothered about all the extra bells and whistles the latter includes.

I have been using IO since the end of 2023 and have only had one instance where Intelligent Octopus failed to tell the zappi to charge our EV. Don't use it that often though, so will give it a try in the next few days to see if I get the error messages you are experiencing.

Double check your account on myenergi's website and make sure you have no scheduled charges setup.
https://myaccount.myenergi.com/schedules

Can you check if your zappi (or myenergi hub, if applicable) has a reliable connection to the internet?

**Comment by AJediIAm** [comment]
I updated the troubleshooting steps. It now mentions that the integration is not compatible with Octopus. Feel free to refine the statement where needed.

**Reply by LeiChat** [reply]
How is it incompatible, please?
Has there been any official comment on the myenergi forums about it, or is it just users discussing it?

I have just enabled IO via Home Assistant, and it appears to have created 'planned dispatches' without issue.
(N.B. the times shown in the screenshot were before I amended it to 10%)

![image](https://github.com/user-attachments/assets/46b84eb9-2d59-4ab9-a0a4-84e7a54ea514)

And the Octopus app on my phone popped a couple of notifications:

![image](https://github.com/user-attachments/assets/44ca5171-6b7e-4de0-aa23-1ac8efcbd98d)

**Reply by AJediIAm** [reply]
I have read the comments and observations here in GitHub, so I added the warning so people are informed that it could be a thing. I have no experience with Octopus myself so I don't know any details. Hopefully others can elaborate and provide more precise expectations of what works and what doesn't.

**Reply by G6EJD** [reply]
The Octopus integration takes control of the Zappi to charge an aEV when their demand rates drop, it is a condition of Octopus that no other integration controls the EV. Many customers have not realised that setting a schedule on the Zappi gets almost immediately cancelled as when not charging Octopus issues frequent stop messages to the Zappi.
So it's not much an incompatibility issue, it's the fact that Octopus requires exclusive Zappi control.
Octopus explains all this in their T&C's and if you don't let Octopus charge your EV they cancel the contract.

**Reply by LeiChat** [reply]
👍 I misread your earlier reply. It's (probably) fine to use ha-myenergi then, we just need to avoid modifying the Charge Mode from HA so it does not interfere with commands being sent by Octopus. I suspect there will be zappi users, who do not use HA, that also encounter the error messages Moondevil is reporting.

**Reply by G6EJD** [reply]
Yes, non-HA users get caught by setting a Zappi schedule and then wonder why the set charge didn't work or why their EV is being charged during the day when there's plenty of solar. Not that it matters as the solar offsets the grid usage, but some users can't cope with that concept!
